src/v/utils/tests/rwlock_test.cc

```diff
@@ -0,0 +1,39 @@
+#include "seastarx.h"
+#include "ssx/semaphore.h"
+#include "utils/rwlock.h"
+
+#include <seastar/core/loop.hh>
+#include <seastar/core/sleep.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+SEASTAR_THREAD_TEST_CASE(test_rwlock_r_blocks_w) {
```

Comment:
nit: may need a test for move semantics.

Reply:
done